### ISSUE TITLE
Keeping the boxes within the limits of the container

### ISSUE THREAD

**Issue by volkanunsal** [STRML/react-grid-layout#56]
I don't understand how the demo keeps the boxes within the container. In my own demo, they never want to stay in their container, even though their x and y values never dip below 0 or exceed the maximum w or h I set for them. Here is my code:

```
onLayoutChange(layout) {
  this.setState({layout: layout});
},

getInitialState() {
  var layout = [
    {i: 1, x: 0, y: 0, minW: 0, minH: 3, maxH: 12, w: 4, h: 3},
    {i: 2, x: 4, y: 0, minW: 0, minH: 3, maxH: 12, w: 4, h: 3},
    {i: 3, x: 8, y: 0, minW: 0, minH: 3, maxH: 12, w: 4, h: 3},
    {i: 4, x: 0, y: 1, minW: 0, minH: 3, maxH: 12, w: 12, h: 3}
  ]
  return {layout: layout};
},
render(){
  return <ReactGridLayout
    className="layout"
    rowHeight={30}
    isResizable={true}
    cols={12}
    layout={this.state.layout}
    onLayoutChange={this.onLayoutChange}
    >
    <div key={1}>1</div>
    <div key={2}>2</div>
    <div key={3}>3</div>
    <div key={4}>3</div>
  </ReactGridLayout>;
}
```

**Comment by volkanunsal** [comment]
Also, some collisions are not getting detected:

![screen shot 2015-05-12 at 7 04 41 pm](https://cloud.githubusercontent.com/assets/151600/7600507/d50616b4-f8d9-11e4-82fd-efa3dc93f9b4.png)

**Comment by liamzebedee** [comment]
+1 I am having the exact same issue - thought it was something to do with my implementation, nope, this is definitely a bug at play. @STRML ? :)

**Comment by liamzebedee** [comment]
Yep I suspect it is something to do with the latest version 0.9.0 - reason is, the Github page (gh-pages branch) hasn't been updated for the latest version, and exists without this bug -- something happened inbetween https://github.com/STRML/react-grid-layout/compare/0.8.3...master

**Comment by STRML** [comment]
I am seeing this too. Working on a fix.

**Comment by STRML** [comment]
You should see this fixed after a reinstall in `0.9.1`.

**Comment by STRML** [comment]
For reference, the issue was an improper optimization in `react-draggable` where it would not adjust the translation of an object if it was being moved back to its initial position. I've removed the optimization - it's not clear it sped anything up, anyway.
